app/models/foreman_fog_proxmox/proxmox.rb

```diff
@@ -154,7 +154,7 @@ def client
       raise ::Foreman::Exception, 'User token expired' if token_expired?(e)
     rescue StandardError => e
       logger.warn("failed to create compute client: #{e}")
-      raise e
+      raise ::Foreman::Exception, "Failed to create compute client: #{e.message}" if @client.nil?
```

Comment:
Can it really occur that the block fails and yet `@client` is not nil?

If so, can we assume that `@client` is a valid/useful object and is it really OK in that case to 'ignore' the occurred `StandardError`?

Reply:
Removed the if condition and it still works, modified the code.